app-admin/rex/rex-9999.ebuild

```diff
@@ -0,0 +1,121 @@
+# Copyright 1999-2017 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+# $Id$
+
+EAPI=6
+
+if [[ ${PV} == 9999 ]]; then
+	GITHUB_USER=RexOps
+	GITHUB_REPO=Rex
+	EGIT_REPO_URI="https://github.com/${GITHUB_USER}/${GITHUB_REPO}.git"
+	EGIT_BRANCH="development-1.x"
+	VCS_ECLASS="git-r3"
+else
+	# This is intentional to stop perl-module.eclass doing magic things when it
+	# shouldn't. Like making ${S} contain "Rex" when the git clone has "rex"
+	DIST_AUTHOR=JFRIED
+	DIST_NAME=Rex
+fi
+
+inherit perl-module ${VCS_ECLASS}
+
+[[ ${PV} == 9999 ]] && SRC_URI=""
+
+DESCRIPTION="(R)?ex is a small script to ease the execution of remote commands"
+
+SLOT="0"
+KEYWORDS=""
+IUSE="test"
+
+DZIL_DEPENDS="
+	dev-perl/Dist-Zilla
+	dev-perl/Dist-Zilla-Plugin-MakeMaker-Awesome
+	dev-perl/Dist-Zilla-Plugin-MetaProvides-Package
+	dev-perl/Dist-Zilla-Plugin-OSPrereqs
+	dev-perl/Dist-Zilla-Plugin-OurPkgVersion
+	dev-perl/Dist-Zilla-Plugin-Test-MinimumVersion
+	dev-perl/Dist-Zilla-Plugin-Test-Perl-Critic
+"
+
+RDEPEND="
+	dev-perl/DBI
+	dev-perl/Devel-Caller
+	dev-perl/Digest-HMAC
+	dev-perl/Digest-SHA1
+	dev-perl/Expect
+	dev-perl/Hash-Merge
+	dev-perl/IO-String
+	dev-perl/IO-Tty
+	dev-perl/IPC-Shareable
+	dev-perl/JSON-XS
+	dev-perl/List-MoreUtils
+	dev-perl/Net-OpenSSH
+	dev-perl/Net-SFTP-Foreign
+	dev-perl/Parallel-ForkManager
+	dev-perl/Sort-Naturally
+	dev-perl/String-Escape
+	dev-perl/TermReadKey
+	dev-perl/Test-Deep
+	dev-perl/Text-Glob
+	dev-perl/URI
+	dev-perl/XML-LibXML
+	dev-perl/XML-Simple
+	dev-perl/libwww-perl
+	dev-perl/YAML
+	virtual/perl-Carp
+	virtual/perl-Data-Dumper
+	virtual/perl-Digest-MD5
+	virtual/perl-Exporter
+	virtual/perl-File-Spec
+	virtual/perl-MIME-Base64
+	virtual/perl-Scalar-List-Utils
+	virtual/perl-Storable
+	virtual/perl-Time-HiRes
+"
+
+DEPEND="
+	${RDEPEND}
+	test? (
+		dev-perl/Test-UseAllModules
+		virtual/perl-File-Temp
+	)
+"
+if [[ ${PV} == 9999 ]]; then
+
+	DEPEND="${DEPEND} ${DZIL_DEPENDS}"
+	src_prepare() {
+		local has_missing modname
+		cd "${S}" || die "Can't enter workdir";
+
+		dzil_version="$(dzil version)" || die "Error invoking 'dzil version'"
+		einfo "Generating CPAN dist with ${dzil_version}"
+		has_missing=""
+		einfo "Checking dzil authordeps"
+		dzil authordeps --missing --versions | while read -r modname; do
+			if [[ -z "${has_missing}" ]]; then
+				has_missing=1
+				eerror "'dzil authordeps' indicates missing build dependencies"
+				eerror "These will prevent building, please report a bug"
+				eerror "Missing:"
+			fi
+			eerror "  ${modname}"
+		done
```

Comment:
Can you clarify?

It is a linebased thing so the defaults of `read` mostly JustWork

But I can switch to the monstrosity:

```bash
<shit here> | while IFS= read -d $'\n' -r i ; do
  <shit here>
done
```